### PR TITLE
pulseeffects: update to 5.0.4

### PR DESCRIPTION
**Pull request by Logarithmus**
- pulseeffects: update to 5.0.4 & remove unneded python deps
- pulseeffects-legacy: remove unneded python deps

~~Blocked by https://github.com/void-linux/void-packages/pull/31666, though magically CI passed, I really don't know why!~~
**UPD**: see https://github.com/void-linux/void-packages/pull/31667#issuecomment-869485860

#### General
- [x] This is a new package and it conforms to the [quality requirements](https://github.com/void-linux/void-packages/blob/master/Manual.md#quality-requirements)

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/pulseeffects-legacy/files/README.voidlinux [hunk 1]
```diff
@@ -0,0 +1,3 @@
+Optional dependencies:
+- lsp-plugins (compressor, equalizer, delay, loudness compensator)
+- zam-plugins (maximizer)
```
Comment (paper42) [inline srcpkgs/pulseeffects-legacy/files/README.voidlinux:3]: I am not sure if this is necessary, all of the information here is mentioned in the program

File: srcpkgs/pulseeffects-legacy/template [hunk 2]
```diff
@@ -4,18 +4,22 @@ version=4.8.5
...
- gst-plugins-good1 pulseaudio python3-gobject python3-scipy"
+ pulseaudio-devel sratom-devel zita-convolver-devel
+ rubberband-devel libsamplerate-devel libsndfile-devel"
+depends="calf gst-plugins-good1"
 short_desc="Sound effects for systems using PulseAudio (legacy)"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/wwmm/pulseeffects"
+changelog="https://github.com/wwmm/pulseeffects/blob/v${version}/CHANGELOG.md"
```
Comment (paper42) [inline srcpkgs/pulseeffects-legacy/template:17]: using master for changelog instead of a tag is probably better

File: srcpkgs/pulseeffects/template [hunk 3]
```diff
@@ -1,20 +1,24 @@
...
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/wwmm/pulseeffects"
+changelog="https://github.com/wwmm/pulseeffects/blob/v${version}/CHANGELOG.md"
 distfiles="https://github.com/wwmm/pulseeffects/archive/v${version}.tar.gz"
-checksum=2e14858918b54bee5f6e4898cc803ae2170b4d624407fef39e0831b6584c4a4f
-python_version=3
+checksum=3fa482e2261fe467e30c05e85a55904806b0dc141834fe8b5cfeffedb6da65e8
+conflicts="pulseeffects-legacy"
```
Comment (ericonr) [inline srcpkgs/pulseeffects/template:19]: Why?

File: srcpkgs/pulseeffects-legacy/files/README.voidlinux (hunk 1 above)
Comment (ericonr) [reply srcpkgs/pulseeffects-legacy/files/README.voidlinux:3]: This isn't necessary and zam-plugins doesn't even exist.

File: srcpkgs/pulseeffects-legacy/template [hunk 4]
```diff
@@ -4,18 +4,22 @@ version=4.8.5
...
 distfiles="https://github.com/wwmm/pulseeffects/archive/v${version}.tar.gz"
 checksum=df1c4c4a9811c62a549822dacde3a9e36233ba3ec58817ae52a236f6181a507c
-python_version=3
 conflicts="pulseeffects"
+
+post_install() {
+	vdoc README.md
```
Comment (paper42) [inline srcpkgs/pulseeffects-legacy/template:23]: Why do you think it should be installed? Which information does it provide?

File: srcpkgs/pulseeffects/files/README.voidlinux [hunk 5]
```diff
@@ -0,0 +1,3 @@
+Optional dependencies:
+- lsp-plugins (compressor, equalizer, delay, loudness compensator)
+- zam-plugins (maximizer)
```
Comment (ericonr) [inline srcpkgs/pulseeffects/files/README.voidlinux:3]: Same.

**Review by ericonr — COMMENTED** [review]
Again, there is such a thing as distro policy. I'm reviewing these because I wanted to be sure they wouldn't go through, but it helps no one to try and change things such as the pkg-config implementation in a per package basis with no justification.

File: srcpkgs/pulseeffects-legacy/template (hunk 2 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template:17]: > using master for changelog instead of a tag is probably better

Why? If upstream releases a new version (tag), then changelog will contain info about this new version, though the package version will remain the same. This creates confusion for users IMO.

File: srcpkgs/pulseeffects-legacy/template (hunk 4 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template:23]: @paper42 
```
rg "vdoc README" srcpkgs | wc -l
739
```

File: srcpkgs/pulseeffects/template (hunk 3 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects/template:19]: Because `pulseeffects-legacy` already has `conflicts="pulseeffects"`.
And both packages have many common files. Though I guess it's possible to rename them to change `pulseeffects` to `pulseeffects-legacy`.

File: srcpkgs/pulseeffects-legacy/template (hunk 2 above)
Comment (paper42) [reply srcpkgs/pulseeffects-legacy/template:17]: changelog is not for end users, it is for maintainers and void contributors. It's useful to look in it when I am updating a package, I want to see all changes in all releases, not just the one the template currently references. Also, having a variable in there is very annoying, I want to click on the link in my terminal and not have to change the version.

Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template:17]: OK, I'll change it

File: srcpkgs/pulseeffects-legacy/template (hunk 4 above)
Comment (paper42) [reply srcpkgs/pulseeffects-legacy/template]: I don't generally see a problem with installing a readme file, some readmes provide useful quick start guides, but I don't see a reason in this case. The information inside is not useful imo for void users. There are some screenshots (which are broken btw), list of effects available (accessible from inside the application, so not useful imo), links to packages for distributions (useless) and other for us useless parts. The only part that could be interesting to users are the packages needed for plugins, but we already established that people can get that information from inside the application.

I think many of the packages which install a README don't have to, but I think that should be solved on a case by case basis when someone touches the package. If you ever update a package and notice it installs a readme that is not useful, I am sure void maintainers wouldn't mind if you removed it.

I am not a maintainer, so I might be wrong, but I don't see any value in installing this particular readme.

Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template]: OK, I see you point, I've already removed those `README`'s from this PR.

File: srcpkgs/pulseeffects/template (hunk 3 above)
Comment (Duncaen) [reply srcpkgs/pulseeffects/template:19]: Adding another conflict does not change anything, it just results in two error messages instead of one.

Comment (Logarithmus) [reply srcpkgs/pulseeffects/template:19]: @Duncaen so `conflicts`works two-way? Interesting, didn't know that. Just tested it and it works. 

P. S. Out of curiosity I'm trying to find out which part of `xbps` is responsible for handling conflicts two-way, but reading this file didn't help much: https://github.com/void-linux/xbps/blob/master/lib/transaction_check_conflicts.c
I would appreciate explaining how does it work.

Comment (Duncaen) [reply srcpkgs/pulseeffects/template]: There is no need to do any "two-way" conflict checks as each package that is installed and has `conflicts=` defined will be checked against installed or to be installed packages.

File: srcpkgs/pulseeffects/template [hunk 6]
```diff
@@ -1,20 +1,18 @@
...
 hostmakedepends="itstool pkg-config gettext"
 makedepends="boost-devel glib-devel gsettings-desktop-schemas-devel
  gst-plugins-bad1-devel gtkmm-devel libebur128-devel lilv-devel
- pipewire-devel python3-gobject-devel sratom-devel zita-convolver-devel
- libsndfile-devel libbs2b-devel"
-depends="calf gsettings-desktop-schemas gst-plugins-bad1
- gst-plugins-good1 pipewire python3-gobject python3-scipy zita-convolver
- gstreamer1-pipewire"
+ pipewire-devel sratom-devel zita-convolver-devel
+ rubberband-devel libsamplerate-devel libsndfile-devel"
```
Comment (paper42) [inline srcpkgs/pulseeffects/template:10]: why did you remove libbs2b-devel? I don't think rubberband-devel is needed, could you double check?

File: srcpkgs/pulseeffects-legacy/template (hunk 2 above)
Comment (paper42) [reply srcpkgs/pulseeffects-legacy/template:17]: ping @Logarithmus? Also, please don't resolve conversations that were not resolved.

Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template]: @paper42 sorry for missing it, fixed now

File: srcpkgs/pulseeffects/template (hunk 6 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects/template]: @paper42 I've removed `libbs2b` because `gst-plugins-bad1` already depend on it, and also Arch package doesn't specify it as a dependency.
```
$ xbps-query -RX libbs2b
audacious-plugins-4.1_4
gst-plugins-bad1-1.18.4_3
ladspa-bs2b-0.9.1_3
libavfilter-4.3.2_1
libbs2b-devel-3.1.0_3
qmmp-1.4.6_1
audacious-plugins-4.1_4
gst-plugins-bad1-1.18.4_3
ladspa-bs2b-0.9.1_3
libavfilter-4.3.2_1
libbs2b-devel-3.1.0_3
qmmp-1.4.6_1
```
`rubberband` is optional, I've removed it.

Comment (Duncaen) [reply srcpkgs/pulseeffects/template]: You removed `libbs2b-devel`, its always better to be explicit about dependencies that are directly used to avoid problems when another dependency drops it.

Comment (paper42) [reply srcpkgs/pulseeffects/template]: > `rubberband` is optional, I've removed it.

just a small note for other reviewers, rubberband is an optional runtime dependency, it's not needed for the build

File: srcpkgs/pulseeffects-legacy/template [hunk 7]
```diff
@@ -7,15 +7,14 @@ build_style=meson
...
- gst-plugins-good1 pulseaudio python3-gobject python3-scipy"
+ pulseaudio-devel sratom-devel zita-convolver-devel
+ libsamplerate-devel libsndfile-devel"
+depends="calf gst-plugins-good1"
 short_desc="Sound effects for systems using PulseAudio (legacy)"
-maintainer="Orphaned <orphan@voidlinux.org>"
+maintainer="Artur Sinila <freesoftware@logarithmus.dev>"
 license="GPL-3.0-or-later"
 homepage="https://github.com/wwmm/pulseeffects"
+changelog="https://github.com/wwmm/pulseeffects/blob/master/CHANGELOG.md"
```
Comment (paper42) [inline srcpkgs/pulseeffects-legacy/template:17]: changelog is usually a link to a raw text file: `https://raw.githubusercontent.com/wwmm/pulseeffects/master/CHANGELOG.md`

Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template:17]: @paper42 https://github.com/void-linux/void-packages/pull/31667#discussion_r659343029
> I want to click on the link in my terminal
Why would you want to open raw text instead of pretty rendered Markdown?

File: srcpkgs/pulseeffects/template (hunk 6 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects/template]: > You removed `libbs2b-devel`, its always better to be explicit about dependencies that are directly used to avoid problems when another dependency drops it.

I've put it back

**Review by paper42 — COMMENTED** [review]
bump the revision in pulseeffects-legacy

File: srcpkgs/pulseeffects-legacy/template (hunk 7 above)
Comment (paper42) [reply srcpkgs/pulseeffects-legacy/template:17]: I don't care about how pretty a changelog is, please follow the convention.

**Comment by paper42** [comment]
Why do you think this is blocked by #31666? I can build it fine without that PR and CI seems to agree.

**Comment by paper42** [comment]
> Why do you think this is blocked by #31666? I can build it fine without that PR and CI seems to agree.

Oh, I see now that libsamplerate-devel  pulls  it in, so  this PR shouldn't be blocked.

File: srcpkgs/pulseeffects-legacy/template (hunk 7 above)
Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template:17]: Yet Another Unwritten Void Distro Policy
OK, 'll change it when reach to the laptop later in the day.

Comment (Logarithmus) [reply srcpkgs/pulseeffects-legacy/template]: @paper42 fixed

**Comment by Logarithmus** [comment]
@paper42 is everything fine now?

**Comment by paper42** [comment]
@Logarithmus can you rebase on top of master?

**Comment by Logarithmus** [comment]
> @Logarithmus can you rebase on top of master?

Done. Though I have no idea why the merge conflicts did appear.

**Comment by paper42** [comment]
> > @Logarithmus can you rebase on top of master?
> 
> Done. Though I have no idea why the merge conflicts did appear.

#31203 It's always necessary to look, you messed up the rebase because you didn't. Also, the PR should be in your notifications because you commented on it.

**Comment by Logarithmus** [comment]
@paper42 I've seen a notification for #31203, but didn't know it modified `pulseeffects*` templates. My bad.

**Comment by paper42** [comment]
Where did the revbump of pulseeffects-legacy go?

**Comment by Logarithmus** [comment]
@paper42 put the revbump back

**Comment by paper42** [comment]
lgtm, I have been using this version of pulseeffects for more than a week without a problem.

but in the meantime, pulseeffects 6 was released and the project was renamed to easyeffects. I think this PR should be merged, because this version improves compatibility with pipewire 0.3.31 and the new version comes with big changes (gtkmm4, not being compatible with pulseeffects 5 presets, gstreamer -> native pipewire filters, change in required and optional libraries, the rename, etc.). and a new PR should handle that. @Logarithmus do you want to make the PR or should I do it?

**Comment by Logarithmus** [comment]
@paper42 feel free to make a new PR, I'm busy with https://github.com/void-linux/void-packages/pull/31706 now.

**Comment by ericonr** [comment]
Ugh I think they broke checksums by moving to a different name.

**Comment by ericonr** [comment]
https://build.voidlinux.org/builders/x86_64-musl_builder/builds/37272/steps/shell_3/logs/stdio
